qiskit_superstaq/compiler_output.py

```diff
@@ -58,12 +56,16 @@ def __repr__(self) -> str:
     def __eq__(self, other: Any) -> bool:
         if not isinstance(other, CompilerOutput):
             return False
-        return (
-            self.circuits == other.circuits
-            and self.seq == other.seq
-            and self.jaqal_programs == other.jaqal_programs
-            and self.pulse_lists == other.pulse_lists
-        )
+
+        if hasattr(self, "circuits") and hasattr(other, "circuits"):
+            circuit_check = self.circuits == other.circuits
+            jaqal_check = self.jaqal_programs == other.jaqal_programs
+            pulse_list_check = self.pulse_lists == other.pulse_lists
```

Comment:
do we want to compare `pulses` here as well?

also given that some of these comparisons can be slow, it's probably better to do something like
```suggestion
            return (
                self.circuits == other.circuits
                and self.jaqal_programs == other.jaqal_programs
                and self.pulse_lists == other.pulse_lists
            )
```
so it will stop as soon as one comparison is False rather than always evaluating all three

Reply:
Done. Forgot about this. Thanks.